ldms/src/ldmsd/ldmsd_config.c

```diff
@@ -1045,27 +1066,37 @@ int ldmsd_plugins_usage(const char *plugname)
 	strncpy(library_path, path, sizeof(library_path) - 1);
 
 	int trc=0, rc = 0;
+	int plugfound = 0;
 	while ((libpath = strtok_r(pathdir, ":", &saveptr)) != NULL) {
 		pathdir = NULL;
 		trc = ldmsd_plugins_usage_dir(libpath, plugname);
 		if (trc)
 			rc = trc;
+		else
+			if (plugname)
+				plugfound = 1;
+	}
+	if (plugname && !plugfound) {
+		fprintf(stderr, "%s: no library in %s for %s\n", APP, path, plugname);
+		strncpy(library_path, path, sizeof(library_path) - 1);
+		while ((libpath = strtok_r(pathdir, ":", &saveptr)) != NULL) {
```

Comment:
I'm not sure I understand how this can work. Line 1070 above has already modified pathdir (adding '\0' after each token), so after that logic runs, there will only be a single entry in pathdir, specifically, the 1st one.

Reply:
the strncpy reinits library_path which is aliased by pathdir.
the aliasing of library_path exists because the code was cloned from new_plugin.

Reply:
Ah, OK, thanks for the explanation @baallan